.github/workflows/ci.yml

```diff
@@ -0,0 +1,56 @@
+name: CI
+
+on: [push, pull_request, release]
+
+jobs:
+  qa:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: pre-commit/action@v3.0.0
+
+  check-links:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: gaurav-nelson/github-action-markdown-link-check@v1
+        name: Check links in markdown files
+        with:
+          use-quiet-mode: "yes"
+          use-verbose-mode: "yes"
+
+  build:
+    name: Build all projects
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install dependencies (Ubuntu)
+        if: matrix.os == 'ubuntu-latest'
+        run: |
+          sudo apt update
+          sudo apt upgrade
+          sudo apt install build-essential cmake ninja-build
+
+      - name: Set up MSVC (Windows)
+        if: matrix.os == 'windows-latest'
+        uses: ilammy/msvc-dev-cmd@v1
+        with:
+          arch: x64
+
+      - name: Install dependencies (macOS)
+        if: matrix.os == 'macos-latest'
+        run: brew install ninja
```

Comment:
My question might be wrong. Please correct me if it is the case.

I guess, all 3 actions viz: Install dependencies (Ubuntu),  Set up MSVC (Windows) and Install dependencies (macOS) will be checked for their if condition and only one will execute depending on the system. 

Is it possible, to rearrange it slightly so that name could be like `Install dependencies` and inside that we checl all 3 conditions. 

In short, I am asking, can we put all conditions in one subheading (-name tag)?

Reply:
It's a good question. I don't know, honestly. The syntax for these files is a little ugly, but I agree it would be nicer. I'll take a look on Monday.

Reply:
I would disagree here. I think that having an step per platform with a clear, simple condition of when to run that step is a cleaner approach than trying to put everything in a single step - even if that were possible - just to have a single “Install dependencies” section.

Reply:
Ok, maybe I'll just leave as is then.